src/OpenTelemetry.Exporter.InMemory/InMemoryExporterLoggingExtensions.cs

```diff
@@ -27,7 +27,27 @@ public static OpenTelemetryLoggerOptions AddInMemoryExporter(this OpenTelemetryL
             Guard.ThrowIfNull(loggerOptions);
             Guard.ThrowIfNull(exportedItems);
 
-            return loggerOptions.AddProcessor(new SimpleLogRecordExportProcessor(new InMemoryExporter<LogRecord>(exportedItems)));
+            var logExporter = new InMemoryExporter<LogRecord>(
+                exportFunc: (in Batch<LogRecord> batch) => ExportLogRecord(in batch, exportedItems));
+
+            return loggerOptions.AddProcessor(new SimpleLogRecordExportProcessor(logExporter));
+        }
+
+        private static ExportResult ExportLogRecord(in Batch<LogRecord> batch, ICollection<LogRecord> exportedItems)
+        {
+            if (exportedItems == null)
+            {
+                return ExportResult.Failure;
+            }
+
+            foreach (var log in batch)
+            {
+                log.BufferLogScopes();
```

Comment:
No. Any log exporter using batching needs to buffer because scopes don't last passed `OnEnd`. An alternative approach to this PR would be to (do something like) add an option to `SimpleLogRecordExportProcessor` to toggle on buffering which the in-memory export could use and any others which might also need simple + buffering (I can't think of why that would be the case but I suppose it is possible).